shell/components/cards/ItemCard.vue

```diff
@@ -0,0 +1,362 @@
+<script setup lang="ts">
+import { PropType, computed } from 'vue';
+import { useStore } from 'vuex';
+import { useI18n } from '@shell/composables/useI18n';
+import LazyImage from '@shell/components/LazyImage';
+
+type ItemCardVariant = 'small' | 'medium';
+
+type Label = {
+  key?: string;
+  text?: string;
+}
+
+type Image = {
+  src: string;
+  alt?: Label;
+};
+
+type Pill = {
+  label: Label;
+  tooltip?: Label;
+};
+
+type Status = {
+  icon: string;
+  color?: string;
+  customColor?: string;
+  tooltip?: Label;
+  handleClick?: () => void;
+}
+
+type Header = {
+  image?: Image;
+  title?: Label;
+  statuses?: Status[];
+}
+
+type ItemValue = Record<string, any>;
+
+const store = useStore();
+const { t } = useI18n(store);
+
+const emit = defineEmits<{( e: 'card-click', value: any): void; }>();
+
+const props = defineProps({
+  id: {
+    type:     String,
+    required: true
+  },
+  value: {
+    type:     Object as PropType<ItemValue>,
+    required: true
+  },
+  header: {
+    type:     Object as PropType<Header>,
+    required: true
+  },
+  content: {
```

Comment:
`content` prop should be optional for when we pass a `item-card-content` template from parent, in that case we don't to pass a content var to ItemCard